python/lsst/ts/observatory/control/auxtel/atcs.py

```diff
@@ -553,25 +556,52 @@ async def shutdown(self):
 
     async def open_dome_shutter(self):
         """Task to open dome shutter and return when it is done.
-        """
 
-        self.rem.atdome.evt_shutterInPosition.flush()
+        Raises
+        ------
+        RuntimeError:
+            If mainDoorState.state is neither OPENED nor CLOSED
+
+        """
 
-        await self.rem.atdome.cmd_moveShutterMainDoor.set_start(
-            open=True, timeout=self.open_dome_shutter_time
+        shutter_pos = await self.rem.atdome.evt_mainDoorState.aget(
+            timeout=self.fast_timeout
         )
 
-        self.rem.atdome.evt_summaryState.flush()
-        # TODO: (2019/12) ATDome should now be returning the command only when
-        # the dome is fully open. I'll keep this here for now for backward
-        # compatibility but we should remove this later, after verifying it
-        # is working DM-24490.
-        task_list = [
-            asyncio.ensure_future(self.check_component_state("atdome")),
-            asyncio.ensure_future(self.wait_for_atdome_shutter_inposition()),
-        ]
+        if shutter_pos.state == ATDome.ShutterDoorState.CLOSED:
 
-        await self.process_as_completed(task_list)
+            self.log.debug("Opening dome shutter...")
+
+            await self.rem.atdome.cmd_moveShutterMainDoor.set_start(
+                open=True, timeout=self.open_dome_shutter_time
+            )
+
+            shutter_pos = ATDome.ShutterDoorState(
+                (
+                    await self.rem.atdome.evt_mainDoorState.aget(
+                        timeout=self.fast_timeout
+                    )
+                ).state
+            )
+
+            if shutter_pos != ATDome.ShutterDoorState.OPENED:
+                self.log.debug(
+                    f"Open shutter command returned while shutter still {shutter_pos!r}."
```

Comment:
why does this happen? is it a bug that needs fixing in ATDome? should we be referencing a ticket number?

Reply:
This (and the next one) is to handle the bug in ATDome. Yes, this should be removed once ATDome is fixed, *but* at the same time, this is not the main code base. It is a patch for "release/0.4", to work with the observatory during the run.

Reply:
ah i missed it was the patch. Sorry